### PR TITLE
Instant Flight Paths and Judgement of command not need CastCustomSpell

### PR DESCRIPTION
**Pull request by zwisus**

### CODE REVIEW

File: src/game/WorldHandlers/Spell.cpp [hunk 1]
```diff
@@ -3795,7 +3795,33 @@ void Spell::TakePower()
...
+        // Spell targets a single enemy
+        if (m_spellInfo->EffectImplicitTargetA[j] == TARGET_CHAIN_DAMAGE ||
+            m_spellInfo->EffectImplicitTargetA[j] == TARGET_CURRENT_ENEMY_COORDINATES)
+        {
+            if (m_caster->GetTypeId() == TYPEID_PLAYER)
+            {
+                if (powerType == POWER_ENERGY)
+                    if (uint64 targetGUID = m_targets.getUnitTargetGuid())
+                        for (std::list<TargetInfo>::iterator ihit = m_UniqueTargetInfo.begin(); ihit != m_UniqueTargetInfo.end(); ++ihit)
+                            if (ihit->targetGUID = targetGUID)
```
Comment (Rochet2) [inline src/game/WorldHandlers/Spell.cpp]: This looks bad. You sure assignment was intended? Did not review code logic, just something that sticks to my eye.

**Comment by zwisus** [comment]
lose Power rage @rochet2

File: src/game/WorldHandlers/World.h [hunk 2]
```diff
@@ -331,6 +331,7 @@ enum eConfigBoolValues
     CONFIG_BOOL_DEATH_BONES_WORLD,
     CONFIG_BOOL_DEATH_BONES_BG,
     CONFIG_BOOL_ALL_TAXI_PATHS,
+    CONFIG_BOLL_INSTANT_TAXI,
```
Comment (Rochet2) [inline src/game/WorldHandlers/World.h]: Need to fix typo: BOLL -> BOOL

**Comment by zwisus** [comment]
@rochet2  thanks

**Comment by Rochet2** [comment]
> lose Power rage rochet2

Im not sure what you mean by this. I was only talking about this part of the code `if (ihit->targetGUID = targetGUID)` where an assignment is made instead of just comparing the values with `==` operator.

**Comment by zwisus** [comment]
@Rochet2 this mean for single enemy,not AOE spell but has error.
FIx:
`    for (uint8 j = 0; j < 3; ++j)
    {
        // Spell targets a single enemy
		if (m_spellInfo->EffectImplicitTargetA[j] == TARGET_CHAIN_DAMAGE ||
			m_spellInfo->EffectImplicitTargetA[j] == TARGET_CURRENT_ENEMY_COORDINATES)
		{
			if (m_caster->GetTypeId() == TYPEID_PLAYER)
			{
				if (powerType == POWER_ENERGY || powerType == POWER_RAGE)
					for (std::list<TargetInfo>::iterator ihit = m_UniqueTargetInfo.begin(); ihit != m_UniqueTargetInfo.end(); ++ihit)
					{
						if (ihit->missCondition != SPELL_MISS_NONE)
						{
							hit = false;
						}
						break;
					}
			}
		}
	}`

**Pull request merged**